### ISSUE TITLE
use web browser like AIS-catcher main app

### ISSUE THREAD

**Issue by g7ruh** [jvde-github/AIS-catcher-for-Android#10]
Jasper 

I have loaded the ap and tested with an RTL-SDR dongle. Rather than use another app to view the data, is it possible to have the web server functionality from AIS-catcher added to the Android version?

If so, will it be possible to use the device GPS to provide location data for the web server map?

It would make it a "one stop shop" for AIS-catcher when mobile.

Thanks

Roger G7RUH

**Comment by g7ruh** [comment]
Jasper,

Yes that is the concept. 

I have tried using a 'Pi as a router for WiFi but I assume that AIS-catcher requires an internet connection to use e.g. JSDelivr and unpkg so AIS-Catcher will not run correctly without the internet? The idea being that I can take a 'Pi to the beach and have WiFi connectivity for a few users to access AIS-catcher web pages on the 'Pi.

I have had your Android app on my list of things to do, so after the above, I gave it a try and thought it would be a far more simple solution anyway, hence the request.

I have a USB C multi port adapter so I can use that for the RTL-SDR and also provide power for phone and dongle.

Happy to test when you are ready. Of course the app may get used far more if it has the kind of usability that the web interface provides.

Roger

**Comment by g7ruh** [comment]
Jasper,

basic is fine!!! Would be good to be able to identify the vessel name. If you are at the beach and showing sonebody the app and functionality, it is nice to point at a vessel and give it a name.

Are map tiles cached locally if previously downloaded from e.g. an earlier session?

Roger

**Comment by g7ruh** [comment]
It will be good enough to try and test like that though

Roger

**Comment by g7ruh** [comment]
Great? Nope, Brilliant is a better description. Nice work. Have you updated the app yet?

Should get to battery life related stuff at the weekend. Initial look suggests about 0.75A at 9V off the phone charger (battery fully charged), with dongle (RTL-SDR V3) using 0.28A at 5V. I am waiting for a [https://www.anandtech.com/show/18944/usbc-power-metering-with-the-chargerlab-km003c-a-google-twinkie-alternative](url) but not due until end of month (all the way from China, where else). 

I will also take it to the beach and provide a suitable photo.

**Comment by g7ruh** [comment]
Is it possible to enable phone GPS to use current location so distance and bearing work too?

**Comment by g7ruh** [comment]
Battery use: (got to it earlier than planned: waiting for a delivery at home)

5V in to usb multiport adapter 0.75A to 0.85A  AIS-catcher running and displaying map.
about 1A when scrolling map.

RTL-SDR V3 uses 0.27A while AIS-catcher is running and 0.07A when AIS-catcher is not running, but dongle plugged in.

Phone uses about 0.45A when AIS-catcher is not running

AIS-catcher CPU use is low when monitoring the CPU use pane from Developer Options GPU Watch. Goes to about 50% when scrolling map (that is expected), but apart from that it does not show a significant difference from when AIS-catcher is not running

**Comment by g7ruh** [comment]
Is there a big difference in consumption for different sample rates?

I will try at the weekend. My thought is that CPU usage is quite low anyway when it is running so a change in sample rates will not make much difference. 

What about GPS use?

Older browsers, good point, I ran a rather old Android phone until my Galaxy Note 20 Ultra. The browser got so slow that I just had to upgrade!

Maybe you just have to live with that. How many users would you miss if you did not support older browsers?

**Comment by g7ruh** [comment]
OK, understand your thinking re sample rate, I will test for that at weekend. Honey-do list takes priority though! May not be too many boats about at W/E .... strong wind warning in these parts, so it will be rather blowy and not good for a long stay at the beach. I may wait for a better day to go portable with the portable AIS-catcher setup.

**Comment by g7ruh** [comment]
Thanks Jasper,

re your lack of real devices for testing, I will run several longer term stability tests (i.e. several hours).

I will also do some tests for sample rate V dongle current use. I will do these on two systems, 1) my test environment which is in the house and has low message rate / sec (less than 1) and 2) also using the main antenna splitter feed where the messages / sec is typically 8 to 12 messages / sec daytime. This weekend it may be less  as the weather is wet and very windy. We do need the rain, tonight is the first evening since I planted the summer bedding plants a few months ago that I have not had to water them. More time to play with radio stuff :)

initial test today suggests sample rate changes makes no difference currnet consumption wise at low message rates. Will repeat as above with high message / sec.

Have a good weekend too.
Roger

**Comment by g7ruh** [comment]
Jasper,

is it possible to make the app full screen or only the web server pages?

Roger

**Comment by g7ruh** [comment]
Yes web client is fine, just wondered if the main app can or will do so. OK if it is on your radar.

Getting on with testing: run the app for 4 hours (connected to power supply, of course) stable, no failures.
Current consumption when 5-6 msg / sec similar to 0.5 msg/sec.
Also same consumption for all scan rate settings when 5-6 msg/sec.
Currently running longer test on main aerial so higher msg / sec

**Comment by g7ruh** [comment]
Jasper,

Some test results from different dongles. Using main system and both test and live connected via VHF splitter, so have the same signal.

The current consumption is the same for all sample rates.

It would be helpful for testing to reset the connect time when clearing other data, but no big deal.

Is it possible to keep the RTL-SDR setting, it reverts to TCP every time you disconnect a dongle. It gives me an error message that it cannot find TCP connection, so the same thing could happen if the dongle is not there at startup. I do like the auto start if dongle detected.

As per the table data, you will see that SN 111 has a very low rate compared with live system and SN 300 has no data. This is because the PPM as calibrated is such that it is out of range partly or totally. I think PPM correction setting is needed to allow for some wide ranges of PPM.

Current was measured at the the dongle (between the multiport usb A and the dongle). I used a USB C multiport adapter to feed power to the phone and dongle.

I cannot attach a .pdf or .ods file, I will email.

Roger

**Comment by g7ruh** [comment]
I have run AIS-catcher on the phone for about 6 hours today and at least 4 yesterday. No issues, problems or crashes so far. It seems quite stable.


Weather still not sutable for a visit to the seaside! So have not verifed works when I move. It gives me distance and bearing so it is working.

**Comment by g7ruh** [comment]
Jasper, am I being silly? I cannot find a App version number anywhere? I have looked.
Roger

**Comment by g7ruh** [comment]
Jasper I repeated the tests with PPM set, used all 4 dongles so comparable with previous tests for SN 444 and SN1000 as they are temp compensated.

SN 300 is a little 'deaf' and never gets as many vessels as the other ones. I use it for testing as it has a huge PPM difference.

All devices showed maximum deviation of +1 to -1 on the Freq Shift (ave PPM / min) graph.

When testing dongles on the 'Pi I have consistently found better performance when gain is set manually and the automatic gain is OFF. 

The method I use to set is: use the App Signal Level min / max dB per min graph. Increase gain until max is over 0dB, then back it off slowly until the max is about 0dB during daytime over an hour or two. That seems to give a good minimum of -40 to -50. If auto gain is on, the range is much less (base is e.g. -35dB) so get less of the weaker signals, I suspect.

The conclusion from these tests is that the Android App is not (yet) as sensitive as the 'Pi app using the same aerial via a splitter.
In percentage terms, the App is about 10% lower. No big deal for occasional use (my use case) the biggest feature is using it portable e.g at the beach or near the docks.

Dongle serial numbers in below tables  444 (RTL-SDR V3), 1000 (FlightAware Orange), 111 (RTL-SDR V2) and at the bottom 300 (RTL-SDR V2)

![image](https://github.com/jvde-github/AIS-catcher-for-Android/assets/5949609/1ab0a526-78b6-42c6-8688-ccd6218a128e)

![image](https://github.com/jvde-github/AIS-catcher-for-Android/assets/5949609/c0b3e52b-72a9-41ef-9628-e5825e970205)


I will email the spreadsheet, Sheet 1 is the main data, Sheet 2 has percentages added.

Data from both tests below:

![image](https://github.com/jvde-github/AIS-catcher-for-Android/assets/5949609/f086df5c-f370-401a-8cfe-c89e0d49bbc3)

I hope that gives you a better idea of performance. Pretty good I would say.

Roger

**Comment by Taxom** [comment]
The auto gain in a dongle not intended to work with narrow band periodic short package digital signal. Therefore, it is trying to adjust the gain by the sum of the signal with the background noise.
